pkg/webhook/nginxingress.go

```diff
@@ -158,10 +168,16 @@ var sarAuthenticate = func(ctx context.Context, lgr logr.Logger, cl client.Clien
 }
 
 func (n *nginxIngressResourceValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
-	lgr := logr.FromContextOrDiscard(ctx).WithValues("resourceName", req.Name, "namespace", req.Namespace, "operation", req.Operation).WithName("nginxIngressResourceValidator")
+	var err error
+	defer func() {
+		metrics.HandleWebhookHandlerMetrics(nginxResourceValidationName, err)
+	}()
+
+	lgr := logr.FromContextOrDiscard(ctx).WithValues("resourceName", req.Name, "namespace", req.Namespace, "operation", req.Operation).WithName(nginxResourceValidationName.LoggerName())
 
 	// ensure user has permissions required
-	cantPerform, err := n.authenticate(ctx, lgr, n.client, req)
+	var cantPerform string
+	cantPerform, err = n.authenticate(ctx, lgr, n.client, req)
```

Comment:
need to set an error if cantPerform != "" if we are incrementing error count on the Denied response.

Alternatively we can add an additional variable to the `metrics.HandleWebhookHandlerMetrics` fn to take in the admission.Response and increment error if it's denied.

Reply:
I'd probably recommend the second approach.

Reply:
I feel like that's a lot easier conceptually than setting errors

Reply:
swapped to using admission response as input to metric handler